auto_backup/model/backup_scheduler.py

```diff
@@ -0,0 +1,425 @@
+            _('Error ,No such database exists'), ['name']
+        )
+    ]
+
+    @api.multi
+    def test_sftp_connection(self):
+        confs = self.search([])
+        # Check if there is a success or fail and write messages
+        messageTitle = ""
+        messageContent = ""
+        for rec in confs:
+            # db_list = self.get_db_list(cr, uid, [], rec.host, rec.port)
+            try:
+                # pathToWriteTo = rec.sftppath
+                conn_success = True
+                ipHost = rec.sftpip
+                portHost = rec.sftpport
+                usernameLogin = rec.sftpusername
+                passwordLogin = rec.sftppassword
+                # Connect with external server over SFTP, so we know sure that
+                # everything works.
+                srv = pysftp.Connection(host=ipHost, username=usernameLogin,
+                                        password=passwordLogin, port=portHost)
+                srv.close()
+                # We have a success.
+                messageTitle = _("Connection Test Succeeded!")
+                messageContent = _(
+                    "Everything seems properly set up for FTP back-ups!")
+            except Exception as e:
+                conn_success = False
+                messageTitle = _("Connection Test Failed!")
+                if len(rec.sftpip) < 8:
+                    messageContent += _(
+                        "\nYour IP address seems to be too short.\n")
+                messageContent += _("Here is what we got instead:\n")
+        if not conn_success:
+            raise except_orm(
+                _(messageTitle), _(
+                    messageContent + "%s") %
+                tools.ustr(e))
+        else:
+            raise Warning(_(messageTitle), _(messageContent))
+
+    @api.model
+    def schedule_backup(self):
+
+        for rec in self.search([]):
+            db_list = self.get_db_list(rec.host, rec.port, rec.securehost)
+            if rec.name in db_list:
+                file_path = ''
+                bkp_file = ''
+                try:
+                    if not os.path.isdir(rec.bkp_dir):
+                        os.makedirs(rec.bkp_dir)
+                    # Create name for dumpfile.
+                    bkp_file = '%s_%s.dump.zip' % (
+                        time.strftime('%d_%m_%Y_%H_%M_%S'),
+                        rec.name)
+                    file_path = os.path.join(rec.bkp_dir, bkp_file)
+                    conn = self.get_connection(
+                        rec.host, rec.port, rec.securehost)
+                    bkp = ''
+                    bkp = execute(
+                        conn, 'dump', rec.adminpassword, rec.name)
+                except:
+                    _logger.info(
+                        _(
+                            "Couldn't backup database %s. "
+                            "Bad database administrator"
+                            "password for server running at http://%s:%s"
```

Comment:
I think this message is in general wrong, because within ```try``` other errors than 'bad password' can happen